src/lib/ddev/config_builder.ts

```diff
@@ -0,0 +1,209 @@
+    this.apiClient = apiClient;
+  }
+
+  public async build(
+    appInstallationId: string,
+    type: string,
+  ): Promise<Partial<DDEVConfig>> {
+    const appInstallation = await this.getAppInstallation(appInstallationId);
+    const systemSoftwares =
+      await this.buildSystemSoftwareVersionMap(appInstallation);
+
+    return {
+      override_config: true,
+      type: await this.determineProjectType(appInstallation, type),
+      webserver_type: "apache-fpm",
+      php_version: this.determinePHPVersion(systemSoftwares),
+      database: await this.determineDatabaseVersion(appInstallation),
+      docroot: await this.determineDocumentRoot(appInstallation),
+      web_environment: [
+        `MITTWALD_APP_INSTALLATION_ID=${appInstallation.shortId}`,
+      ],
+    };
+  }
+
+  private async determineDocumentRoot(inst: AppInstallation): Promise<string> {
+    const appVersion = await this.getAppVersion(
+      inst.appId,
+      inst.appVersion.desired,
+    );
+
+    if (appVersion.docRootUserEditable && hasCustomDocumentRoot(inst)) {
+      return stripLeadingSlash(inst.customDocumentRoot);
+    }
+
+    return stripLeadingSlash(appVersion.docRoot);
+  }
+
+  private async determineProjectType(
+    inst: AppInstallation,
+    type: string,
+  ): Promise<string> {
+    if (type !== "auto") {
+      return type;
+    }
+
+    if (inst.appId === typo3Installer.appId) {
+      return "typo3";
+    }
+
+    if (inst.appId === wordpressInstaller.appId) {
+      return "wordpress";
+    }
+
+    if (inst.appId === shopware6Installer.appId) {
+      return "shopware6";
+    }
+
+    if (inst.appId === drupalInstaller.appId) {
+      const version = await this.getAppVersion(
+        inst.appId,
+        inst.appVersion.desired,
+      );
+
+      const [major] = version.externalVersion.split(".");
+      return `drupal${major}`;
+    }
+
+    throw new Error(
+      "Automatic project type detection failed. Please specify the project type manually by setting the `--override-type` flag.",
+    );
+  }
+
+  private async determineDatabaseVersion(
+    inst: AppInstallation,
+  ): Promise<DDEVDatabaseConfig | undefined> {
+    const isPrimary = (db: LinkedDatabase) => db.purpose === "primary";
+    const primary = (inst.linkedDatabases || []).find(isPrimary);
+
+    if (primary?.kind === "mysql") {
+      const r = await this.apiClient.database.getMysqlDatabase({
+        mysqlDatabaseId: primary.databaseId,
+      });
+      assertStatus(r, 200);
+
+      return {
+        type: "mysql",
+        version: r.data.version,
+      };
+    }
+
+    return undefined;
+  }
+
+  private determinePHPVersion(
+    systemSoftwareVersions: SystemSoftwareVersions,
+  ): string | undefined {
+    if (!("php" in systemSoftwareVersions)) {
+      return undefined;
+    }
+
+    const version = systemSoftwareVersions["php"];
+    const [major, minor] = version.split(".");
+
+    return `${major}.${minor}`;
```

Comment:
Split on `.` and the join with `.`? 🤔 or is this to drop the patch level? 

Could be more intuitive if extracted `removePatchLevel(version)` 🤷

Reply:
> or is this to drop the patch level?

Exactly this; refactoring incoming...